From 6cab761b36240185db72ab481c65270dd7b00ab3 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Thu, 01 Oct 2020 16:15:01 +0300
Subject: [PATCH] [SW-13981] efuse: add failure notification

Add error notification indicating a failure with reading the eFuse data

Change-Id: I004d87328e8a1813bc332119b1e7daae42864c27
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/common/efuse.c b/common/efuse.c
index b7f79aa..909c582 100644
--- a/common/efuse.c
+++ b/common/efuse.c
@@ -8,6 +8,8 @@
 #include <unistd.h>
 #include <psoc_efuse_regs.h>
 #include <psoc_dft_efuse_regs.h>
+#include <preboot.h>
+#include <boot_if.h>
 #include "efuse.h"
 #include "common.h"
 #include "preboot.h"
@@ -54,12 +56,14 @@
 	u32 read_val;
 
 	if (word >= FUSE_WORDS_PER_BANK) {
+		cpu_boot_err_clrset(CPU_BOOT_ERR0_EFUSE_FAIL, true);
 		preboot_puts("Invalid efuse word offset\n");
 		return -1;
 	}
 
 	ret = preboot_fuse_init();
 	if (ret == -2) {
+		cpu_boot_err_clrset(CPU_BOOT_ERR0_EFUSE_FAIL, true);
 		preboot_puts("Efuse not initialized\n");
 		return ret;
 	}
